plotly/presentation_objs/presentation_objs.py

```diff
@@ -0,0 +1,1199 @@
+            'src': text_or_url + '.embed?link=false',
+            'style': {'height': height,
+                      'left': left,
+                      'position': 'absolute',
+                      'top': top,
+                      'width': width}
+        }
+    elif boxtype == 'CodePane':
+        props = {
+            'language': 'python',
+            'source': text_or_url,
+            'style': {'fontFamily': fontFamily,
+                      'fontSize': 13,
+                      'height': height,
+                      'left': left,
+                      'margin': 0,
+                      'position': 'absolute',
+                      'textAlign': 'left',
+                      'top': top,
+                      'width': width},
+            'theme': 'tomorrowNight'
+        }
+
+    # update props and style attributes
+    for item in props_attr.items():
+        props[item[0]] = item[1]
+    for item in style_attr.items():
+        props['style'][item[0]] = item[1]
+
+    child = {
+        'children': children_list,
+        'id': id,
+        'props': props,
+        'type': boxtype
+    }
+
+    if boxtype == 'Text':
+        child['defaultHeight'] = 36
+        child['defaultWidth'] = 52
+        child['resizeVertical'] = False
+    if boxtype == 'CodePane':
+        child['defaultText'] = 'Code'
+
+    return child
+
+
+def _percentage_to_pixel(value, side):
+    if side == 'left':
+        return WIDTH * (0.01 * value)
+    elif side == 'top':
+        return HEIGHT * (0.01 * value)
+    elif side == 'height':
+        return HEIGHT * (0.01 * value)
+    elif side == 'width':
+        return WIDTH * (0.01 * value)
+
+
+def _return_box_position(left, top, height, width):
+    values_dict = {
+        'left': left,
+        'top': top,
+        'height': height,
+        'width': width,
+    }
+    for key in values_dict.keys():
+        if isinstance(values_dict[key], str):
+            if values_dict[key][-2:] != 'px':
+                raise exceptions.PlotlyError(
+                    NEEDED_STYLE_ERROR_MESSAGE
+                )
+            try:
+                var = float(values_dict[key][: -2])
+            except ValueError:
+                raise exceptions.PlotlyError(
+                    NEEDED_STYLE_ERROR_MESSAGE
+                )
+
+        else:
+            var = _percentage_to_pixel(values_dict[key], key)
+        values_dict[key] = var
+
+    return (values_dict['left'], values_dict['top'],
+            values_dict['height'], values_dict['width'])
+
+
+def _remove_extra_whitespace_from_line(line):
+    while line.startswith('\n') or line.startswith(' '):
+        line = line[1:]
+    while line.endswith('\n') or line.endswith(' '):
+        line = line[: -1]
+    return line
+
+
+def _list_of_slides(markdown_string):
+    if not markdown_string.endswith('\n---\n'):
+        markdown_string += '\n---\n'
+
+    text_blocks = re.split(
+        '\n--\n|\n---\n|\n----\n|\n-----\n|\n------\n', markdown_string
```

Comment:
Would a regex like `\n-{2,}\n` be more generic? Check out https://regex101.com/ for testing regexes